### ISSUE TITLE
Validate Ubuntu 18.04 Bionic Beaver for v2.0

### ISSUE THREAD

**Issue by jgreat** [rancher/rancher#13888]
The docker install script fails when trying to install on Ubuntu 18.04.
https://releases.rancher.com/install-docker/17.03.sh

Doesn't look like the Bionic repo for docker-ce has any packages, much less the older 17.03.2 package.
https://download.docker.com/linux/ubuntu/dists/bionic/stable/

**Rancher versions:**
rancher/rancher: v2.0.2

**Comment by superseb** [comment]
See https://github.com/rancher/rancher/issues/13222#issuecomment-394307237

**Comment by superseb** [comment]
Other fix when using Ubuntu 18.04:

https://github.com/rancher/rancher/issues/13782
```
Ubuntu 18.04 uses local DNS cache nameserver which breaks kube-dns which is used to query DNS. Steps to fix this are:

sudo systemctl mask systemd-resolved
rm -f /etc/resolv.conf
echo "nameserver 8.8.8.8" > /etc/resolv.conf
```

**Comment by lucevers** [comment]
snap docker could be a solution but there is also a blocking issue:

 [](https://forum.snapcraft.io/t/snap-docker-and-rke-kubernetes/5845)

**Comment by superseb** [comment]
https://github.com/docker/docker-snap/issues/31

**Comment by superseb** [comment]
Using Docker 18.05 (currently testing for Ubuntu 18.04) will show:

```
[workerPlane] Failed to bring up Worker Plane: Failed to start [kubelet] container on host [159.65.193.41]: Error response from daemon: linux mounts: Could not find source mount of /var/lib/docker
```

Caused by https://github.com/moby/moby/issues/37032

**Comment by Napsty** [comment]
I can confirm that the workaround (mask systemd-resolved) mentioned by @superseb works. 
Rancher 2.0.2, Ubuntu 18.04.

**Comment by Rgils** [comment]
Please, try to avoid to "hardcode" 8.8.8.8 and get system dns like in #13504 with: 
```
sudo rm /etc/resolv.conf
sudo ln -s /run/systemd/resolve/resolv.conf /etc/resolv.conf
```

**Comment by twn346** [comment]
https://github.com/docker/docker-ce/releases/tag/v18.06.1-ce

@superseb you mention here: https://github.com/rancher/rancher/issues/13222#issuecomment-394307237 
That the install scripts will be updated once Docker V18.06 was released.
Since Docker v18.06.1-ce was released 6 days ago, and v18.06.0-ce was released on July 18, can we please get an update to the install scripts?

**Comment by cprivite** [comment]
I think they're actually waiting for Kubernetes to support 18.06. Right now it's still only validated on "1.11.2 to 1.13.1 and 17.03.x" : https://kubernetes.io/docs/setup/release/notes/

**Comment by cprivite** [comment]
Then, once Kubernetes supports a version of Docker that is stable and works on Ubuntu 18.04, then Rancher can be updated to work with Ubuntu 18.04.

**Comment by mayconfsbrito** [comment]
It's possible to install docker 17.03 in Ubuntu 18.04 (Bionic) by binary package. I've created an ansible role to do this, you can access by this link:

https://galaxy.ansible.com/mayconfsbrito/ansible_docker_bin

Anyway, I will post here the task of this role to guide you in this process, because I needed to configure some steps (like storage driver, docker group) that aren't explicit in the documentation.

I used the documentation below:

Install from binary:
https://docs.docker.com/v17.09/engine/installation/linux/docker-ce/binaries/#install-static-binaries

Post install:
https://docs.docker.com/install/linux/linux-postinstall/

**Comment by cprivite** [comment]
You still have the issue of Docker itself not publishing or supporting putting 17.03 onto Ubuntu 18.04 which is a non-starter for many companies (mine included). 
I know I can *make* 17.03 run on Ubuntu 18.04, but I want the officially supported, "thousands of people will be using the same config you are" version before I go live on it. And Rancher is being developed to be a stable enterprise solution so unfortunately working around things here isn't going to work, we're just going to have to wait and use 16.04 in the meantime.

**Comment by mayconfsbrito** [comment]
I don't think that installing docker via binary package is workaround. It's just one more solution.

The problem is neither of the docker repository, but the k8s that won't update until now.

**Comment by xiaods** [comment]
rancher 2.1 i also came across this issue. only one host, install the rancher control plane , and k8s cluster , but the finally it report" Alert: Node xx is not active rancher"

docker  version
```
➜  ~ docker version
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:24:51 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:23:15 2018
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by xiaods** [comment]
following @Rgils's comments and ways, i also need do this step to resolve my problems:

```bash
systemctl restart systemd-resolved
```

**Comment by xiaods** [comment]
now this issue can close asap.

**Comment by CalvinHartwell** [comment]
@xiaods it is still broken I believe, I am hitting issues when attempting to deploy 2.1.x+ on Bionic. Actually I don't get any GUI at all, but I need to continue testing.

**Comment by Napsty** [comment]
@CalvinHartwell This might be of interest to you.

We had a major issue last week on our Rancher 2.0 staging environment which ran on Bionic, too (but with a supported Docker version). All of a sudden, in the middle of the night, out of nowhere, the staging cluster stopped working. Even Rancher UI started failing (timeouts, error 502 from LB).

We have now switched all nodes of all Rancher clusters to Xenial and deployed a new Rancher 2.1 cluster. Not only does it "feel" more stable, the UI definitely reacts quicker (this could be the new 2.1 version though). 

We ran the staging environment on Bionic since August. Yet this issue out of the blue forced us to decide to switch (back) to Xenial, as it is officially documented in the OS requirements of Rancher 2.0.

**Comment by cprivite** [comment]
Actually, the rancher documentation says "Rancher is supported on the following operating systems and their subsequent releases." and then goes on to list Ubuntu 16.04. This implies that 18.04 is supported as well, even though it seems clear that it isn't.

**Comment by superseb** [comment]
It's supported on the listed OS' and their non-major releases, this ticket exists to validate this release, as long as it's open, it's not validated. When k8s 1.12 is our next default version, we can validate Ubuntu 18.04 with a properly released Docker version on this release and which is supported in k8s 1.12 as well.

**Comment by cprivite** [comment]
That is not what people usually mean when they say "subsequent". That implies 16.04 is the minimum and anything later is good too. I would choose different language, but hey you can keep getting people confused and installing it on 18.04 if you want. :)

**Comment by cjellick** [comment]
Need to discuss with @cloudnautique @ibuildthecloud and @sangeethah for a plan how to fix and then validate this

**Comment by cjellick** [comment]
just looking at the comments, this looks like it will definitely not be super straightfoward

**Comment by superseb** [comment]
The biggest hold-up on this was that Kubernetes only supported Docker up to 17.03 at the time Ubuntu 18.04 was released. In k8s 1.12 the support for Docker versions extended to: `1.11.1, 1.12.1, 1.13.1, 17.03, 17.06, 17.09, 18.06.` (https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.12.md#external-dependencies) which now includes a version where Docker publishes a package for (https://download.docker.com/linux/ubuntu/dists/bionic/pool/stable/amd64/).

Regarding the systemd-resolved issue, we added https://github.com/rancher/rke/issues/827 a long time ago to automatically use the correct `resolv.conf` file with upstream nameservers.

So basically this means we only have to run QA on Ubuntu 18.04 with Docker 18.06 and k8s 1.12 and higher.

**Comment by sangeethah** [comment]
@jiaqiluo We need to certify Ubuntu 18.04 Bionic Beaver os with docker 18.06.

Execute current set of automation runs on the following configuration:
1 control
3 etcd
3 worker nodes

**Comment by jiaqiluo** [comment]
Validated on Rancher: master 12/20

Setup: 
- Installed Rancher:master by both single installation and HA installation.
- A custom cluster with the following configuration: 1 control, 3 etcd, 3 worker nodes

Spec for both Rancher server and nodes in clusters:
- Ubuntu 18.04.1 LTS
- Docker 18.06.1-ce, build e68fc7a
- (nodes) Kubenete version: 1.12.3
 
used this command to install docker:
```
curl -sSL https://releases.rancher.com/install-docker/18.06.sh |sh
```

**Tests finished on Rancher:**
- Test v3 API - Execute existing Automation runs (Covers network tests relating to inter host communication and being able to access Ingress from outside cluster)
- Storage test - persistent volumes with NFS

**Test finished on RKE:**
- Backup/Restore etcd - Restore cluster from backups - https://rancher.com/docs/rke/v0.1.x/en/etcd-snapshots/
```
1. Create a 1 node RKE cluster
2. Create few workloads
3. Take a snapshot
4. Create a new 1 RKE node cluster
5. Copy the snapshot taken in step3 to the new node
6. Restore the snapshot on the new node
7. Should be able to see the workloads created in step2 in the new cluster.
8. Should be able to create new workloads

**Comment by Napsty** [comment]
@jiaqiluo please provide some additional information during the setup process, e.g.:

- which rke version
- which helm version
- which rancher version

this should help clarify things. thank you.
